### PR TITLE
Enhance search page functionality and style.

### PR DESCRIPTION
**Pull request by zrideaux**
+ Improve functionality of dropdown menus.
  + Add function to hide all dropdown menus when called.
  + Dropdowns are now toggled when clicked. 
+ Add width to dropdown menu to avoid unusual wrapping.
+ Improve style of search filter buttons. 
+ Wrap listings in anchor elements to make them clickable.
  + Note: Exact format of this can be changed when implementing backend.

### CODE REVIEW

File: src/main/webapp/search.css [hunk 1]
```diff
@@ -16,7 +16,9 @@ h2 {
 #search-keyword-container {
   align-items: center;
   display: flex;
+  margin-bottom: 0.5em;
   padding: 5px;
+
```
Comment (kev1n80) [inline src/main/webapp/search.css:21]: Please delete empty space

File: src/main/webapp/search.css [hunk 2]
```diff
@@ -52,10 +59,37 @@ main {
...
+  border: 1px solid #BDBDBD;
+  border-radius: 15px;
+  outline: none;
+  transition: border 0.5s; 
+}
+
+.dropdown-box button:hover {
+  background-color: #EEEEEE;
+  border: 1px solid #000000;
+}
```
Comment (kev1n80) [inline src/main/webapp/search.css:73]: I love this styling addition!

File: src/main/webapp/search.html [hunk 3]
```diff
@@ -11,18 +11,18 @@
...
         <input id="search-keyword" type="text" placeholder="Search by name">
       </div>
 
       <div id="search-options-container">
-        <div id="search-location">
+        <span id="search-location">
           <label for="search-location-input">Location</label>
           <input id="search-location-input" type="text" placeholder="Location">
-        </div>
+        </span>
```
Comment (kev1n80) [inline src/main/webapp/search.html:22]: Why did you decide to switch from div to span? (curious)

File: src/main/webapp/search.js [hunk 4]
```diff
@@ -1,15 +1,27 @@
...
 /**
- * Reveal a specified dropdown menu when called. 
+ * Toggle a specified dropdown menu when called. 
  */
-function showDropdown(menuName) {
-  // Hide any other open dropdown menus
+function toggleDropdown(menuName) {
+  const dropdownMenu = document.getElementById(menuName);
+  let menuIsClosed = (dropdownMenu.style.display === 'none' 
+                      || dropdownMenu.style.display === '');
```
Comment (kev1n80) [inline src/main/webapp/search.js:8]: I have a toggleDisplay method that you should leverage here! Also, I recommend replacing getComputedStyle(dropdownMenu, null).display over dropdownMenu.style.display.
getComputedStyle(dropdownMenu, null) 
accounts for the css applied to an element whereas 
dropdownMenu.style
does not.

File: src/main/webapp/search.html (hunk 3 above)
Comment (zrideaux) [reply src/main/webapp/search.html:22]: I wanted all of the settings to be inline, and it made more sense to me to just use span as opposed to changing the display mode of #search-location 
image

File: src/main/webapp/search.js (hunk 4 above)
Comment (zrideaux) [reply src/main/webapp/search.js:8]: Oh, awesome, I will swap over to that now. Thanks! Also, where can I find the toggleDisplay method?

**Pull request merged**